### ISSUE TITLE
Tensorflow JS Converter Can't Find Expected FusedBatchNorm Input 

### ISSUE THREAD

**Issue by ByrdOfAFeather** [tensorflow/tfjs#5691]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow.js):
 Yes. I have used the following script to convert a graph model to a .pb format as I could not find official documentation on how to do this:  

```
import tensorflow as tf

meta_path = 'test.meta' # Your .meta file
output_node_names = ['pose/locref_pred/block4/BiasAdd', 'pose/part_pred/block4/BiasAdd']    # Output nodes

with tf.compat.v1.Session() as sess:
    # Restore the graph
    saver = tf.compat.v1.train.import_meta_graph(meta_path)

    # Load weights
    saver.restore(sess,tf.compat.v1.train.latest_checkpoint('checkpoint_folder'))

    # Freeze the graph
    frozen_graph_def = tf.compat.v1.graph_util.convert_variables_to_constants(
        sess,
        sess.graph_def,
        output_node_names, variable_names_blacklist=[n.name for n in sess.graph_def.node if "FusedBatchNorm" in n.name])

    # Save the frozen graph
    with open('_graph.pb', 'wb') as f:
      f.write(frozen_graph_def.SerializeToString())
```

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
  Kubuntu 21.04
- TensorFlow.js installed from (npm or script link):
  Installed from pip 
- Tensorflow.js Converter Version:
  3.9.0

**Describe the current behavior**
When using this command :
```tensorflowjs_converter --input_format=tf_frozen_model --output_node_names='pose/locref_pred/block4/BiasAdd','pose/part_pred/block4/BiasAdd' output_graph.pb output --skip_op_check```

I get
```2021-10-04 11:07:07.150459: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory
2021-10-04 11:07:07.150484: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2021-10-04 11:07:13.137106: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:1137] Optimization results for grappler item: graph_to_optimize
  debug_stripper: debug_stripper did nothing. time = 0.369ms.
  model_pruner: Graph size after: 1130 nodes (-530), 1201 edges (-530), time = 29.391ms.
  constant_folding: Graph size after: 1088 nodes (-42), 1147 edges (-54), time = 105.765ms.
  arithmetic_optimizer: Graph size after: 1088 nodes (0), 1147 edges (0), time = 52.367ms.
  dependency_optimizer: Graph size after: 1064 nodes (-24), 1123 edges (-24), time = 23.379ms.
  model_pruner: Graph size after: 1064 nodes (0), 1123 edges (0), time = 10.789ms.
  constant_folding: Graph size after: 1064 nodes (0), 1123 edges (0), time = 42.342ms.
  arithmetic_optimizer: Graph size after: 1064 nodes (0), 1123 edges (0), time = 41.601ms.
  dependency_optimizer: Graph size after: 1064 nodes (0), 1123 edges (0), time = 26.46ms.
  debug_stripper: debug_stripper did nothing. time = 2.391ms.
  model_pruner: Graph size after: 1064 nodes (0), 1123 edges (0), time = 15.009ms.
  constant_folding: Graph size after: 1064 nodes (0), 1123 edges (0), time = 40.475ms.
  arithmetic_optimizer: Graph size after: 1064 nodes (0), 1123 edges (0), time = 33.552ms.
  dependency_optimizer: Graph size after: 1064 nodes (0), 1123 edges (0), time = 20.708ms.
  model_pruner: Graph size after: 1064 nodes (0), 1123 edges (0), time = 12.907ms.
  constant_folding: Graph size after: 1064 nodes (0), 1123 edges (0), time = 44.195ms.
  arithmetic_optimizer: Graph size after: 1064 nodes (0), 1123 edges (0), time = 37.797ms.
  dependency_optimizer: Graph size after: 1064 nodes (0), 1123 edges (0), time = 25.408ms.

WARNING:tensorflow:Didn't find expected Conv2D or DepthwiseConv2dNative input to 'resnet_v1_101/block4/unit_1/bottleneck_v1/conv2/BatchNorm/FusedBatchNormV3'
WARNING:tensorflow:Didn't find expected Conv2D or DepthwiseConv2dNative input to 'resnet_v1_101/block4/unit_2/bottleneck_v1/conv2/BatchNorm/FusedBatchNormV3'
WARNING:tensorflow:Didn't find expected Conv2D or DepthwiseConv2dNative input to 'resnet_v1_101/block4/unit_3/bottleneck_v1/conv2/BatchNorm/FusedBatchNormV3'
```

The warnings here appear to be problematic. When loading the files into a browser, I get 

``tf.loadGraphModel('http://127.0.0.1:5000/download/model.json')
Promise {<pending>}
util.ts:105 Uncaught (in promise) Error: Based on the provided shape, [3,3,512,512], the tensor should have 2359296 values but has 2300067``

**Describe the expected behavior**

Either the warnings should not exist (i.e. the FusedBatchNorm is converted properly), or the model should be able to load in the browser without the batch norm layers

**Standalone code to reproduce the issue**
https://github.com/ByrdOfAFeather/tfjs_bug

**Comment by rthadur** [comment]
@ByrdOfAFeather  you mentioned you are using tfjs-converter 3.9 but in your code there is reference to older tfjs version https://github.com/ByrdOfAFeather/tfjs_bug/blob/master/templates/val_model.html#L6 , is this intended ?

**Comment by rthadur** [comment]
and also Frozen models are not supported in latest versions. Thank you

**Comment by ByrdOfAFeather** [comment]
> @ByrdOfAFeather you mentioned you are using tfjs-converter 3.9 but in your code there is reference to older tfjs version https://github.com/ByrdOfAFeather/tfjs_bug/blob/master/templates/val_model.html#L6 , is this intended ?

Yes so this is my bad - I was referring to my python install of tensorflowjs and didn't check the version I was using on the web, I just tried it with tfjs@3.9.0 and receive the same error when trying to load in the model. 



> and also Frozen models are not supported in latest versions. Thank you

If this is the case, is there anyway to take a snapshot.meta, snapshot.index, shapshot.data and use it with tensorflowjs? I've being trying to find a way to convert these into a Tensorflow 2 style model (they were generated with tensorflow.compat.v1) but can't seem to find a way to do this anywhere.

**Comment by rthadur** [comment]
I have no idea , sorry. @pyu10055  ++ for any help . Thank you

**Comment by pyu10055** [comment]
@ByrdOfAFeather I tried your model, the loading did not generate any error for me. But I cannot execute the model since it contains no input placeholder nodes, can you confirm that you have selected the correct output nodes for freezing the model?


```html

<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Document</title>
</head>
<body>
    <script src="https://cdn.jsdelivr.net/npm/@tensorflow/tfjs"></script>
    <script>
        window.addEventListener("DOMContentLoaded", main);

        const MODEL_URL = "./model.json";

        async function main() {
            const model = await tf.loadGraphModel(MODEL_URL);
            console.log(model);
        }
    </script>
</body>
</html>
```

**Comment by pyu10055** [comment]
@ByrdOfAFeather maybe you can share screenshot of the network tab in the chrome developer tools while you loading the model.
It is likely that you failed to download the binary files.
If you have python3, you can use `python -m http.server` to start a web work, but all your files needs to be under the directory where you run the command.

**Comment by ByrdOfAFeather** [comment]
@pyu10055 
![Screenshot_20211009_221143](https://user-images.githubusercontent.com/23299767/136678761-d4691dac-3086-42fb-bab2-9a3e0085c1fc.png)
![Screenshot_20211009_221146](https://user-images.githubusercontent.com/23299767/136678762-3c2e7077-82f4-4252-b88d-0ff29b6927c1.png)
![Screenshot_20211009_221151](https://user-images.githubusercontent.com/23299767/136678763-3c173ec9-c8b9-46a1-90df-9882f2b27184.png)

All files seem to be downloaded successfully

**Comment by ByrdOfAFeather** [comment]
> @ByrdOfAFeather this is the model.json file, not the weight files. Weight files are binary. Please check your weight files, usually their size are 4MB.

I see, there was an issue in the repository I submitted for the bug. I fixed it and can now load the model. Now I also see that there are no placeholder elements. It is unclear to me how to solve this, I believe I have selected the correct output nodes.

**Comment by ByrdOfAFeather** [comment]
I went back and used an export function in the software that I was using rather than using my own export function, I know cna load the model and have input nodes. https://github.com/DeepLabCut/DeepLabCut/blob/ce681b4073ab1a0bd5823f892ebe0f9c47a0f7f6/deeplabcut/pose_estimation_tensorflow/export.py#L234 For future reference. 

Thanks for helping - I'm going to go ahead and close since this seems to be resolved.
